bokeh/server/session.py

```diff
@@ -63,6 +64,10 @@ def _needs_document_lock(func):
                 self._pending_writes = []
                 try:
                     result = func(self, *args, **kwargs)
+                    if inspect.iscoroutine(result):
```

Comment:
I think we want `isawaitable` here (a bit more precise)

https://docs.python.org/3/library/inspect.html#inspect.isawaitable